src/main/java/com/petros/bring/bean/factory/AnnotationBeanFactory.java

```diff
@@ -128,6 +120,27 @@ private <T> T createBean(BeanDefinition beanDefinition) {
         }
     }
 
+    private <T> T createDependentBean(Class<T> dependsOnClass) {
+        checkCircularDependency(dependsOnClass.getSimpleName());
+        Object o = this.registry.getBeanDefinitionsByType(dependsOnClass).stream()
+                .findFirst()
+                .map(this::create)
+                .orElseThrow(
+                        () -> new BeanCreationException("Could not create bean with type: " + dependsOnClass.getName())
+                );
+        circularDependencies.remove(dependsOnClass.getSimpleName());
+        return dependsOnClass.cast(o);
+    }
+
+    private void checkCircularDependency(String name) {
+        if (circularDependencies.contains(name)) {
+            circularDependencies.add(name + "!");
```

Comment:
didnt get this, adding duplicated name but with exclamation?

Reply:
It's been done in order to make exception message more understandable. There will be something like "Could not initialize application. Circular dependencies are found:  A -> B -> C -> D -> A!